Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM python:3.8-slim-buster as base
+
+
+FROM base as production
+WORKDIR /app
+
+RUN pip install poetry \
+  && poetry config virtualenvs.create false
+
+COPY poetry.lock pyproject.toml /app/
```

Comment:
You could move common setup into the base. 
i.e. WORKDIR, installing poetry, copying poetry.lock and pyproject.toml